sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/TableIdentifierParserSuite.scala

```diff
@@ -23,39 +23,39 @@ class TableIdentifierParserSuite extends SparkFunSuite {
-    "true", "truncate", "update", "user", "values", "with", "regexp", "rlike",
-    "bigint", "binary", "boolean", "current_date", "current_timestamp", "date", "double", "float",
-    "int", "smallint", "timestamp", "at", "position", "both", "leading", "trailing", "extract")
+  // It is recommended to list them in alphabetical order.
+  val hiveNonReservedKeyword = Array("add", "admin", "after", "all", "alter", "analyze", "any",
+    "archive", "array", "as", "asc", "at", "authorization", "autocommit", "before", "between",
+    "bigint", "bigint", "binary", "boolean", "both", "bucket", "buckets", "by", "cascade",
+    "change", "cluster", "clustered", "clusterstatus", "collection", "columns", "comment",
+    "compact", "compactions", "compute", "concatenate", "continue", "cost", "create", "cube",
+    "current_date", "current_timestamp", "cursor", "data", "databases", "date", "datetime", "day",
+    "dbproperties", "decimal", "deferred", "defined", "delete", "delimited", "dependency", "desc",
+    "describe", "directories", "directory", "disable", "distribute", "double", "drop", "enable",
+    "escaped", "exclusive", "exists", "explain", "export", "external", "extract", "false", "fetch",
+    "fields", "file", "fileformat", "first", "float", "for", "format", "formatted", "functions",
+    "grant", "group", "grouping", "hold_ddltime", "hour", "idxproperties", "ignore", "import",
+    "in", "index", "indexes", "inpath", "inputdriver", "inputformat", "insert", "int", "into",
+    "is", "isolation", "items", "jar", "key_type", "keys", "last", "lateral", "leading", "level",
+    "like", "limit", "lines", "load", "local", "location", "lock", "locks", "logical", "long",
+    "mapjoin", "materialized", "metadata", "minus", "minute", "month", "msck", "no_drop", "none",
+    "noscan", "null", "nulls", "of", "offline", "offset", "option", "order", "out", "outer",
+    "outputdriver", "outputformat", "overwrite", "owner", "partition", "partitioned", "partitions",
+    "percent", "pivot", "plus", "position", "pretty", "principals", "procedure", "protection",
+    "purge", "query", "range", "read", "readonly", "reads", "rebuild", "recordreader",
+    "recordwriter", "regexp", "reload", "rename", "repair", "replace", "replication", "restrict",
+    "revoke", "rewrite", "rlike", "role", "roles", "rollup", "row", "rows", "schemas", "second",
+    "serde", "serdeproperties", "server", "set", "sets", "shared", "show", "show_database",
+    "skewed", "smallint", "snapshot", "sort", "sorted", "ssl", "statistics", "stored",
+    "streamtable", "string", "struct", "table", "tables", "tblproperties", "temporary",
+    "terminated", "timestamp", "tinyint", "to", "touch", "trailing", "transaction", "transactions",
+    "trigger", "true", "truncate", "unarchive", "undo", "uniontype", "unlock", "unset", "unsigned",
+    "update", "uri", "use", "user", "utc", "utctimestamp", "values", "view", "while", "with",
+    "work", "write", "year")
```

Comment:
put them in alphabet order and remove some repeated words, including:
`"int", "smallint", "double", "float", "timestamp", "binary", "boolean", "both", "current_date", "current_timestamp", "date"`